elections/uk/every_election.py

```diff
@@ -0,0 +1,248 @@
+                self.post_object = post_extra.base
+                self.post_created = False
+            except PostExtra.DoesNotExist:
+                self.post_object = Post.objects.create(
+                    label=label,
+                    organization=self.organization_object,
+                )
+                post_extra = PostExtra.objects.create(
+                    base=self.post_object,
+                    slug=slug,
+                )
+                self.post_created = True
+
+            self.post_object.role = role
+            self.post_object.label = label
+            self.post_object.organization = self.organization_object
+            self.post_object.save()
+            post_extra.party_set = self.get_or_create_partyset()[0]
+            post_extra.save()
+        return (self.post_object, self.post_created)
+
+    def get_or_create_post_election(self, parent):
+        if hasattr(self, 'post_extra_object'):
+            self.post_extra_created = False
+        else:
+            # First, set up the Post and Election with related objects
+            self.get_or_create_post()
+
+            # Make sure we're creating an ID for the right parent
+            assert self.parent == parent['election_id'], "{} != {}".format(
+                self.parent, parent['election_id']
+            )
+
+            # Note that we create the election of the parent
+            parent.get_or_create_election()
+
+            # Get the winner count
+            if self['division']:
+                winner_count = self['division']['seats_total']
+            else:
+                # TODO this might not always be true, get the count from EE
+                winner_count = 1
+
+            self.post_extra_object, self.post_extra_created = \
+                PostExtraElection.objects.update_or_create(
+                    postextra=self.post_object.extra,
+                    election=parent.election_object,
+                    defaults={
+                        'ballot_paper_id': self['election_id'],
+                        'winner_count': winner_count,
+                    }
+                )
+        return (self.post_extra_object, self.post_extra_created)
+
+
+class EveryElectionImporter(object):
+    def __init__(self, query_args=None):
+        self.EE_BASE_URL = getattr(
+            settings, "EE_BASE_URL", "https://elections.democracyclub.org.uk/")
+
+        self.election_tree = {}
+        if not query_args:
+            query_args = {
+                'current': 'True',
+            }
+        self.query_args = query_args
+
+
+    def build_election_tree(self):
+        """
+        Get all current elections from Every Election and build them in to
+        a tree of IDs
+        """
+
+        url = "{}api/elections/".format(self.EE_BASE_URL)
+        prams = urlencode(self.query_args)
+        url = "{}?{}".format(url, prams)
+        while url:
+            req = requests.get(url)
```

Comment:
I think it would be useful to `raise_for_status()` here so that if the EE API call fails we don't accidentally cause unexpected behaviour by trying to parse an error response.